design/namedspaces.go

```diff
@@ -0,0 +1,27 @@
+package design
+
+import (
+	d "github.com/goadesign/goa/design"
+	a "github.com/goadesign/goa/design/apidsl"
+)
+
+var _ = a.Resource("namedspaces", func() {
+	a.BasePath("/namedspaces")
+
+	a.Action("show", func() {
+		a.Routing(
+			a.GET("/:userName/:spaceName"),
+		)
+		a.Description("Retrieve space (as JSONAPI) for the given user name and space name.")
+		a.Params(func() {
+			a.Param("userName", d.String, "User name of the owner of the space")
```

Comment:
Why not have an `a.MinLenght(1)` here? Then you can get rid of the checks in the namespace controller https://github.com/almighty/almighty-core/pull/701/files#diff-c05c495dcd3b2bc599451164f1fab446R27.